client/evilwallet/output_manager.go

```diff
@@ -174,15 +174,31 @@ func (o *OutputManager) UpdateOutputsFromTxs(txIDs []string) error {
 
 // GetOutput returns the Output of the given outputID.
 func (o *OutputManager) GetOutput(outputID ledgerstate.OutputID) (output *Output) {
```

Comment:
What I don't like about this function is that we make an API call each time, so during the spam we send API request 100*number of inputs per second, and we usually have all needed informations stored in *evilwallet.Output and can simply recreate an input like 
```
evilInput := wallet.GetUnspentOutput()
outBalance := getIotaColorAmount(evilInput.Balance)
out := ledgerstate.NewSigLockedColoredOutput(evilInput.Balance, evilInput.Address)
input := out.SetID(evilInput.OutputID)
```

Reply:
We only request api if the provided unspent output is not in the wallets (line 177), I think this could be useful in some cases, users can provide valid input even though it doesn't belong to one of the wallet we managed.

Reply:
ok, I did not notice that it is not used always, then it's alright